duckduckgo/Android
# Feature/migrate to androidx

Task/Issue URL: https://app.asana.com/0/414730916066338/885014473436729
Tech Design URL: 
CC: 

**Description**:
Upgrade to `AndroidX`, and upgrade a bunch of dependencies along with it. Many dependencies weren't compatible with `AndroidX` at their existing version, so increasing their version was required to let our app upgrade.

Upgrading various libraries resulted in more stricter tooling checks so there are additional code tidy ups to fix many of these new errors/warnings.

**Steps to test this PR**:
1. Ensure the project builds (all dependencies resolved etc...)
1. Ensure all unit tests pass for you
1. App requires a **thorough** testing; there is a huge footprint of code changes here, and some new libraries to boot. So there's a lot of places where a regression could have crept in.

If possible, it would be good to get both @brindy and @subsymbolic to at least test the app out, even if only one of you will do the actual cod review. 🙏 

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
